src/main/java/io/cdap/plugin/successfactors/source/config/SuccessFactorsPluginConfig.java

```diff
@@ -0,0 +1,371 @@
+    return this.paginationType;
+  }
+
+  /**
+   * Checks if the call to SuccessFactors service is required for metadata creation.
+   * condition parameters: ['host' | 'serviceName' | 'entityName' | 'username' | 'password']
+   * - any parameter is 'macro' then it returns 'false'
+   *
+   * @return boolean flag as per the check
+   */
+  public boolean isSchemaBuildRequired() {
+    return !(containsMacro(UNAME) || containsMacro(PASSWORD) || containsMacro(BASE_URL) || containsMacro(ENTITY_NAME));
+  }
+
+  /**
+   * @return the schema of the dataset
+   */
+  @Nullable
+  public Schema getSchema(FailureCollector collector) {
+    try {
+      return Strings.isNullOrEmpty(schema) ? null : Schema.parseJson(schema);
+    } catch (IOException e) {
+      collector.addFailure("Invalid schema: " + e.getMessage(), null)
+        .withConfigProperty(NAME_SCHEMA);
+    }
+    // if there was an error that was added, it will throw an exception, otherwise, this statement will not be executed
+    throw collector.getOrThrowException();
+  }
+
+  public void validatePluginParameters(FailureCollector failureCollector) {
+    validateMandatoryParameters(failureCollector);
+    validateBasicCredentials(failureCollector);
+    LOG.debug("Validating the advanced parameters.");
+    validateEntityParameter(failureCollector);
+    failureCollector.getOrThrowException();
+  }
+
+  /**
+   * Validates the mandatory parameters.
+   *
+   * @param failureCollector {@code FailureCollector}
+   */
+  private void validateMandatoryParameters(FailureCollector failureCollector) {
+
+    IdUtils.validateReferenceName(getReferenceName(), failureCollector);
+    if (SuccessFactorsUtil.isNullOrEmpty(getBaseURL()) && !containsMacro(BASE_URL)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("SAP SuccessFactor Base URL");
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(BASE_URL);
+    }
+    if (SuccessFactorsUtil.isNotNullOrEmpty(getBaseURL()) && !containsMacro(BASE_URL)) {
+      if (HttpUrl.parse(getBaseURL()) == null) {
+        String errMsg = ResourceConstants.ERR_INVALID_BASE_URL.getMsgForKey("SAP SuccessFactor Base URL");
+        failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(BASE_URL);
+      }
+    }
+    if (SuccessFactorsUtil.isNullOrEmpty(getEntityName()) && !containsMacro(ENTITY_NAME)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("Entity Name");
+      failureCollector.addFailure(errMsg, COMMON_ACTION).withConfigProperty(ENTITY_NAME);
+    }
+  }
+
+  /**
+   * Validates the credentials parameters.
+   *
+   * @param failureCollector {@code FailureCollector}
+   */
+  private void validateBasicCredentials(FailureCollector failureCollector) {
+
+    if (SuccessFactorsUtil.isNullOrEmpty(getUsername()) && !containsMacro(UNAME)) {
+      String errMsg = ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("SAP SuccessFactors Username");
```

Comment:
use variable for all getMsgForKey parameters

Reply:
Added constants for all strings. 4dbd00c9806f47df7f78cf411834bc8c7a613f94